arcgis-ios-sdk-samples/Maps/Manage operational layers/MMLLayersViewController.swift

```diff
@@ -15,150 +15,149 @@
 import UIKit
 import ArcGIS
 
-protocol MMLLayersViewControllerDelegate: AnyObject {
-    func layersViewControllerWantsToClose(_ layersViewController: MMLLayersViewController, withDeletedLayers layers: [AGSLayer])
-}
-
 class MMLLayersViewController: UITableViewController {
-    var layers: NSMutableArray!
-    var deletedLayers: [AGSLayer]!
     
-    weak var delegate: MMLLayersViewControllerDelegate?
+    /// A change handler that calls back with the operational layers.
+    var onChange: (([AGSLayer]) -> Void)?
+    
+    /// Every layer on the map or that could be added to the map.
+    var removedLayers: [AGSLayer] = []
+    /// The layers attached to the map in back-to-front order.
+    var operationalLayers: [AGSLayer] = []
+    
+    /// The operational layers in reverse order so the topmost layer is first.
+    private var operationalLayersForTable: [AGSLayer] {
+        set {
+            if newValue != operationalLayersForTable {
+                operationalLayers = newValue.reversed()
+                // the user edited the layers in the table so run the change handler
+                onChange?(operationalLayers)
+            }
+        }
+        get{
+            return operationalLayers.reversed()
+        }
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        self.tableView.isEditing = true
-    }
-    
-    func dataSourceIndexForIndexPath(_ dataSource: NSMutableArray, indexPath: IndexPath) -> Int {
-        return dataSource.count - indexPath.row - 1
+        // enable the editing UI
+        tableView.isEditing = true
     }
     
-    enum Section: Int {
-        case added
-        case removed
+    /// A convenience type for the table view sections.
+    private enum Section: Int {
+        case operational, removed
         
-        init(_ rawValue: Int) {
-            self.init(rawValue: rawValue)!
+        var label: String {
```

Comment:
This is only used in `titleForHeaderInSection`, so I think it makes more sense to leave the switch statement where it was.

Reply:
I understand experimenting. You may have thought that the title would be used from more than one place. But in the end, it wasn't. And it isn't likely to be used anywhere else. So leaving that change in your branch left me scratching my head. Why go through the extra work of moving the switch statement, wrapping it in a property, and then calling that property from the original location? It didn't make sense.

When you're proofreading your changes before submitting a PR, try to keep an eye out for unnecessary changes like this and removing the custom initializer. Taking the time to revert those unnecessary changes before review reduces the amount of work the reviewer has to do and avoids any potential confusion. Does that make sense?